packages/core/test/controls/numericInputTests.tsx

```diff
@@ -0,0 +1,736 @@
+            inputField.simulate("blur");
+
+            expect(onConfirmSpy.calledOnce).to.be.true;
+            expect(onConfirmSpy.calledWith("3 + a")).to.be.true;
+        });
+
+        it("accepts a numeric value", () => {
+            const component = mount(<NumericInput value={10} />);
+            const value = component.state().value;
+            expect(value).to.equal("10");
+        });
+
+        it("accepts a string value", () => {
+            const component = mount(<NumericInput value={"10"} />);
+            const value = component.state().value;
+            expect(value).to.equal("10");
+        });
+    });
+
+    describe("Keyboard interactions", () => {
+
+        const simulateIncrement = (component: ReactWrapper<any, {}>, mockEvent?: IMockEvent) => {
+            let mergedMockEvent = mockEvent || {};
+            mergedMockEvent.keyCode = Keys.ARROW_UP;
+            mergedMockEvent.which = Keys.ARROW_UP;
+
+            const inputField = component.childAt(0).find("input");
+            inputField.simulate("keyDown", mergedMockEvent);
+        };
+
+        const simulateDecrement = (component: ReactWrapper<any, {}>, mockEvent?: IMockEvent) => {
+            let mergedMockEvent = mockEvent || {};
+            mergedMockEvent.keyCode = Keys.ARROW_DOWN;
+            mergedMockEvent.which = Keys.ARROW_DOWN;
+
+            const inputField = component.childAt(0).find("input");
+            inputField.simulate("keyDown", mergedMockEvent);
+        };
+
+        runInteractionSuite("Press '↑'", "Press '↓'", simulateIncrement, simulateDecrement);
+    });
+
+    describe("Mouse interactions", () => {
+
+        const simulateIncrement = (component: ReactWrapper<any, {}>, mockEvent?: IMockEvent) => {
+            const incrementButton = component.childAt(1).childAt(0);
+            incrementButton.simulate("click", mockEvent);
+        };
+
+        const simulateDecrement = (component: ReactWrapper<any, {}>, mockEvent?: IMockEvent) => {
+            const decrementButton = component.childAt(1).childAt(1);
+            decrementButton.simulate("click", mockEvent);
+        };
+
+        runInteractionSuite("Click '+'", "Click '-'", simulateIncrement, simulateDecrement);
+    });
+
+    describe("Value bounds", () => {
+
+        describe("if no bounds are defined", () => {
+
+            it("enforces no minimum bound", () => {
+                const component = mount(<NumericInput />);
+
+                const decrementButton = component.childAt(1).childAt(1);
+                decrementButton.simulate("click", { shiftKey: true });
+                decrementButton.simulate("click", { shiftKey: true });
+
+                const newValue = component.state().value;
+                expect(newValue).to.equal("-20");
+            });
+
+            it("enforces no maximum bound", () => {
+                const component = mount(<NumericInput />);
+
+                const incrementButton = component.childAt(1).childAt(0);
```

Comment:
there should be a more reliable way to find the Button that doesn't rely on traversing the tree.

perhaps `component.find(Button).first/last()`?